pkg/telemetryquerylanguage/functions/tqlcommon/README.md

```diff
@@ -3,13 +3,28 @@
 The following functions can be used in any implementation of the Telemetry Query Language.  Although they are tested using [pdata](https://github.com/open-telemetry/opentelemetry-collector/tree/main/pdata) for convenience, the function implementation only interact with native Go types or types defined in the [tql package](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/pkg/telemetryquerylanguage/tql).
 
 Factory Functions
+- [Concat](#concat)
 - [IsMatch](#ismatch)
 
 Functions
 - [set](#set)
 - [replace_match](#replace_match)
 - [replace_pattern](#replace_pattern)
 
+## Concat
+
+`Concat(...values)`
```

Comment:
Let's add a parameter for `delimiter`.  The `delimiter` should be a string and will be added in-between each value.  If such a value isn't needed, the `delimiter` should be set to `""`

Reply:
I think this would make sense given that the TQL should get an expression parser to concatenate strings in the future, and this could make a factory function useful for cases where the user wants to join string values with a delimiter.

However, I think if we go that route, we should rename the function to `Join` so it is in line with how other languages name this operation. Additionally, I think the method signature would be more obvious as `Join(values, delimiter)` where `values` is a list of values. This would require modifications to the grammar, so I've opened up an issue to discuss whether that might make sense here: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/13391. What do you think?

Reply:
I'm good renaming the Factory Function to Join.  

Yes we should definitely support Slices in the grammar, thanks for opening that issue.  If you want to hold off on this PR until Slices are added thats ok, but I think we could also do `Join(delimiter, values)` in the meantime.

Reply:
I was thinking we would have to wait unless we want to introduce a breaking change after adding slices. Is this a place where that would be okay?

Reply:
The component/tql are still alpha so breaking changes are ok.  And if we need to in the future there are ways to deprecate functions over several releases to reduce the burden.